core/commands/id.go

```diff
@@ -97,15 +99,17 @@ EXAMPLE:
 			return errors.New(offlineIdErrorMessage)
 		}
 
-		p, err := n.Routing.FindPeer(req.Context, id)
-		if err == kb.ErrLookupFailure {
+		// We need to actually connect to run identify.
+		err = n.PeerHost.Connect(req.Context, peer.AddrInfo{ID: id})
```

Comment:
This is a fine conceptual change. Just a note that this doesn't actually do anything as the DHT (our only PeerRouting system) will connect to the peer, and will not return any addresses if it fails. https://github.com/libp2p/go-libp2p-kad-dht/blob/fc3558cc35274f5d1727dad8201f821df1ee6317/routing.go#L675-L682

Reply:
Hm. I thought I had some trouble where we weren't actually _waiting_ for the connect event to finish.

But honestly, we should do this anyways. We shouldn't assume that the router will always end up connecting to the target peer.